### PR TITLE
Initial commit of files for DR6 build and release

### PR DESCRIPTION
**Pull request by tapastro**
Contains conversion from setup to pyproject, publish action, and removal of test script due to lack of truth files.

### CODE REVIEW

**Review by stscirij — COMMENTED** [review]
Hi Tyler - looks like your PR is missing a bunch of recent commits to main, e.g. the tests in tests/test_wrapper.py have been updated to use test data stored on Box.  However, the github action that runs the tests (.github/workflows/python_testing.yml) will need to be changed as it calls python setup.py install instead of pip install

**Comment by tapastro** [comment]
Hi Robert, we had a short discussion about the testing files during the DP meeting yesterday - my understanding was that tests for this build wouldn't be useful, as the truth files were generated from this commit (effectively - the truth files are DR6 outputs). I can update the file rather than delete it, I just wasn't sure if there was value to including it in DR6. Certainly, a future build would include these updates.

**Comment by jotaylor** [comment]
@stscirij we will include the testes in the forthcoming DR7 build, but as Tyler says, the tests were not working at the time of DR6 so the easiest solution was to just remove them. We could retcon things, but that may be more work than its worth for a prior release build. Thoughts?

File: src/ullyses/calibrate_stis_data.py
Comment (stscirij) [inline src/ullyses/calibrate_stis_data.py:828]: Typo here - direcotry

Comment (stscirij) [reply src/ullyses/calibrate_stis_data.py:828]: Although this is already merged...

**Review by stscirij — APPROVED** [review]
It's hard to disentangle the changes in this PR from those that were updates to the main branch since the release-dr6 branch was made.  I'll approve this assuming the main changes for the PR are:

- adding publish-to-pypi action
- removing tests
- setup.py -> pyproject.toml
- move code from ullyses to src/ullyses

**Comment by stscirij** [comment]
Once this gets merged I'll see if it's worth trying to put the tests back

**Comment by tapastro** [comment]
Ideally that's all that's been done - the commit history is a bit messy because I started the repo branch from dr5 - it was easier to make the release-dr6 branch by cloning the existing dr5 than making a new one. Sorry 😬

File: src/ullyses/calibrate_stis_data.py
Comment (jotaylor) [reply src/ullyses/calibrate_stis_data.py:828]: I can add this in my misc. DR7 updates branch 👍

Comment (jotaylor) [reply src/ullyses/calibrate_stis_data.py:828]: Wait, sorry, I realized this was already fixed in later code. It will be a relic bug of the past then 😆

**Pull request merged**
